src/tests/ftest/pool/evict_test.py

```diff
@@ -20,128 +20,258 @@
   portions thereof marked with this legend must also reproduce the markings.
-'''
+"""
 from __future__ import print_function
 
-import os
-import traceback
-
-from avocado.utils import process
 from apricot import TestWithServers
+from daos_api import DaosApiError
+from general_utils import get_container
+from test_utils import TestPool
+from conversion import c_uuid_to_str
+import ctypes
+import uuid
+
 
-class EvictTest(TestWithServers):
+class EvictTests(TestWithServers):
     """
+
     Tests DAOS client eviction from a pool that the client is using.
 
     :avocado: recursive
     """
 
-    def test_evict(self):
+    def connected_pool(self, hostlist, targets=None):
+        """Connect to pool.
+
+        Args:
+            hostlist (list): list of daos server nodes
+            targets (list): List of targets for pool create
+        Returns:
+            TestPool (object)
+
         """
-        Test evicting a client to a pool.
+        pool = TestPool(self.context, self.log)
+        pool.get_params(self)
+        if targets is not None:
+            pool.target_list.value = targets
+        # create pool
+        pool.create()
+        # Check that the pool was created
+        status = pool.check_files(hostlist)
+        if not status:
+            self.log.error("Invalid pool - pool data not detected on servers")
+            return None
+        # Connect to the pool
+        status = pool.connect(1)
+        if not status:
+            self.log.error("Pool connect failed or already connected")
+            return None
+        # Return connected pool
+        return pool
+
+    def evict_badparam(self, test_param):
+        """Connect to pool, connect and try to evict with a bad param.
+
+        Args:
+            test_param (str): either invalid UUID or bad server name
+
+        Returns:
+            TestPool (bool)
 
-        :avocado: tags=pool,poolevict,quick
         """
-        size = self.params.get("size", '/run/tests/sizes/size1gb/*')
-        setid = self.params.get("setname", '/run/tests/setnames/validsetname/*')
-        connectperm = self.params.get("perms",
-                                      '/run/tests/connectperms/permro/*')
+        # setup pool and connect
+        self.pool = self.connected_pool(self.hostlist_servers)
+
+        self.log.info(
+            "Pool UUID: %s\n Pool handle: %s\n Server group: %s\n",
+            self.pool.uuid, self.pool.pool.handle.value, self.pool.name)
 
+        if test_param == "BAD_SERVER_NAME":
+            # Attempt to evict pool with invald server group name
+            # set the server group name directly
+            self.pool.pool.group = ctypes.create_string_buffer(test_param)
+            self.log.info(
+                "Evicting pool with invalid Server Group Name: %s", test_param)
+        elif test_param == "invalid_uuid":
+            # Attempt to evict pool with invald UUID
+            bogus_uuid = str(uuid.uuid4())
```

Comment:
If we want to be super cautious and avoid a one-in-a-million false failure:
```suggestion
            bogus_uuid = self.pool.uuid
            while bogus_uuid == self.pool.uuid:
                bogus_uuid = str(uuid.uuid4())
```